### ISSUE TITLE
LoHa/LoKr merge in train mode bakes stochastic rank_dropout masks into merged weights

### ISSUE THREAD

**Issue by Sravanjangam** [huggingface/peft#3586]
Hi PEFT team! :wave: One more from the same read-only audit series as #3581–#3583 (@ `5d602fdb`): merging LoHa/LoKr in train mode bakes **random dropout masks** into the merged weights, making `merge_adapter()` non-reproducible. AI-assisted audit per AGENTS.md; mechanism verified by the submitting human; tracker searched first.

**Environment:** peft `0.20.1.dev0` @ `5d602fdb` · transformers `5.15.1` · torch `2.13.0` CPU · Python 3.12 · macOS arm64

### Problem

For LoHa/LoKr, `get_delta_weight` applies a stochastic per-row dropout when `self.training and rank_dropout > 0` (`src/peft/tuners/lycoris_utils.py:132-147`, used by `loha/layer.py:209-218` and `lokr/layer.py:279-286`). That code path is shared between training-time forward (where the randomness is intended) and the merge paths — but nothing puts the model into eval mode before merging: neither `merge_adapter()` (`tuners_utils.py:1269-1296`) nor `_unload_and_optionally_merge` checks/sets mode, and `get_peft_model()` leaves modules in train mode.

Consequence: calling `merge_and_unload()` after training without an explicit `.eval()` first (a very common post-Trainer state) permanently folds an independent random mask + rescaling per layer into the base weights. The merged model matches neither the trained adapter nor any reproducible function of it. Calling it twice gives two different merged models.

For contrast, LoRA's own `get_delta_weight` (`lora/layer.py:1005-1033`) is deterministic — LoHa/LoKr are the outliers here.

### Reproduction sketch

```python
# LoHaConfig(rank_dropout=0.25, ...), train briefly, then WITHOUT .eval():
m.merge_adapter(); w1 = <capture q_proj weight>; m.unmerge_adapter()
m.merge_adapter(); w2 = <capture q_proj weight>; m.unmerge_adapter()
# w1 != w2  ← merge result depends on RNG state
```

### Suggested fix

Make merge-time deltas deterministic regardless of module mode — e.g. compute the delta with dropout explicitly disabled (a parameter on `get_delta_weight`, or a temporary eval-mode switch inside `merge()`/`unmerge()`), keeping training-time forward behavior untouched. Since unmerge recomputes the delta to invert a merge, determinism also restores the merge→unmerge inverse property in this configuration.

Happy to prepare the PR with regression tests (two merges with fixed seed must produce identical weights while `training=True`). Thanks! 🙏
